CONTRIBUTING.md

```diff
@@ -1,20 +1,191 @@
+- Enter your credit/debit card information for billing purposes, opt for the free basic support.
+- Agree to the AWS Customer Agreement and Service Terms, complete the registration by clicking `"Create Account and Continue"`, verify your phone number via text or call, confirm your email address following the instructions in the confirmation email, and finally sign in to access your new AWS account using your email and password.
 
-- Explain your process.
\ No newline at end of file
+<sub>[Back to Table of Contents](#table-of-contents)</sub>
+***
+
+## **Installing Terraform**
+
+We are using the [HashiCorp Tutorials](https://developer.hashicorp.com/terraform/tutorials/aws-get-started/install-cli) for installing terraform.
+
+### **MacOS installation**
+
+In the guide we are using [homebrew](https://brew.sh) for the installation, make sure you have homebrew installed on your machine.
+
+- We will first install the `hashicorp tap` a repository of all the terraform homebrew packages, use the below command in your terminal.
+    
+    ```bash
+    brew tap hashicorp/tap
+    ```
+- Then install terraform with brew
+
+    ```bash
+    brew install terraform
+    ```
+    now confirm your installation with `terraform -help`.
+
+### **Windows installation**
+
+For windows installation we will use [choclatey](https://chocolatey.org) a free and open-source package management system for Windows.
+
+- Open the `PowerShell` terminal as an administrator and install the terraform package with 
+
+    ```bash
+    choco install terraform
+    ```
+- To confirm its installation close the current terminal and open a new `PowerShell` terminal as a normal user and use `terraform -help`.
+
+### **Linux installation**
+
+To install Terraform on your Linux distribution, visit the [HashiCorp Tutorial](https://developer.hashicorp.com/terraform/tutorials/aws-get-started/install-cli) and follow the provided instructions for your specific distribution. After installation, verify Terraform by running `terraform -help` in your terminal.
+
+<sub>[Back to Table of Contents](#table-of-contents)</sub>
+***
+
+## **Installing Terraform docs**
+
+To install Terraform docs we will be using Docker, terraform-docs can be run as a container by mounting a directory with `.tf` files in it and run the following command:
+
+    ```bash
+    docker run --rm --volume "$(pwd):/terraform-docs" -u $(id -u) quay.io/terraform-docs/terraform-docs:0.17.0 markdown /terraform-docs
+    ```
+If `output.file` is not enabled for this module, generated output can be redirected back to a file:
+
+    ```bash
+    docker run --rm --volume "$(pwd):/terraform-docs" -u $(id -u) quay.io/terraform-docs/terraform-docs:0.17.0 markdown /terraform-docs > doc.md
+    ```
+**Note:** Docker tag `latest` refers to latest stable released version and `edge` refers to HEAD of `master` at any given point in time. And any named version tags are identical to the official GitHub releases without leading `v`.
```

Comment:
```suggestion
Follow the Terraform docs [installation guide](https://terraform-docs.io/user-guide/installation/)
```
I think it can be as simple as this...no need for Docker unless they want to